onnxruntime/core/framework/tensor.cc

```diff
@@ -27,7 +27,10 @@ Tensor::Tensor(MLDataType p_type, const TensorShape& shape, std::shared_ptr<IAll
     ORT_THROW("shape.Size() must >=0");
 
   void* p_data = nullptr;
-  if (shape_size > 0) {
+  if (shape.NumDimensions() && shape_size) {
+    p_data = allocator->Alloc(shape);
+  }
+  if (!p_data && shape_size > 0) {
```

Comment:
I don't it is OK to skip CalcMemSizeForArray. But since it is `Alloc(shape)` note the **shape** here. This use the new interface . And the default impl return a `nullptr`. If an allocator decide to support this interface, then it is the allocator to handle it.

Since the interface is not settled. I want to leave it for future discussion.